### ISSUE TITLE
Running native automation test in Xcode

### ISSUE THREAD

**Issue by aqrc** [leancodepl/patrol#1614]
Hi there! 
First of all thank you for your work!

I have read [the article about Patrol 2.0](https://leancode.co/blog/patrol-2-0-improved-flutter-ui-testing) and wanted to reproduce what is done in [the video (running tests in Xcode)](https://youtu.be/SphJ-s6hQGk). Spent a day on it but without success.

I configured everything following the [documentation](https://patrol.leancode.co/getting-started/native) and example tests are working fine when I am running them with `patrol test` command.  However, when I run `RunnerUITests` in Xcode I get either a white screen or "Test starting..." screen in simulator and never ending "Testing..." status in Xcode. 

This is output from Xcode console:
```
2023-08-09 01:56:33.325331+0300 RunnerUITests-Runner[84789:7569242] [SceneConfiguration] Info.plist contained no UIScene configuration dictionary (looking for configuration named "(no name)")
2023-08-09 01:56:33.325656+0300 RunnerUITests-Runner[84789:7569242] [SceneConfiguration] Info.plist contained no UIScene configuration dictionary (looking for configuration named "(no name)")
2023-08-09 01:56:33.335812+0300 RunnerUITests-Runner[84789:7569242] Running tests...
2023-08-09 01:56:38.125049+0300 RunnerUITests-Runner[84789:7569242] PatrolServer: INFO: PATROL_PORT is null, falling back to default (8081)
2023-08-09 01:56:38.125202+0300 RunnerUITests-Runner[84789:7569242] PatrolServer: INFO: PatrolServer constructor called
2023-08-09 01:56:38.125282+0300 RunnerUITests-Runner[84789:7569242] PatrolServer: INFO: Fatal error: PATROL_ENABLED flag is not defined
2023-08-09 01:56:38.125818+0300 RunnerUITests-Runner[84789:7569641] Server loop done, error: (null)
2023-08-09 01:56:38.125861+0300 RunnerUITests-Runner[84789:7569242] PatrolAppServiceClient: created, port: 8082
    t =      nans Checking existence of `"Allow" Button`
    t =      nans Open com.example.app.dev
    t =      nans     Launch com.example.app.dev
    t =      nans         Setting up automation session
    t =      nans         Wait for com.example.app.dev to idle
2023-08-09 01:56:42.422711+0300 RunnerUITests-Runner[84789:7569242] [TraitCollection] Class CKBrowserSwitcherViewController overrides the -traitCollection getter, which is not supported. If you're trying to override traits, you must use the appropriate API.
```

I thought on `PatrolServer: INFO: Fatal error: PATROL_ENABLED flag is not defined`, so I tried to add `-D PATROL_ENABLED` everywhere but it didn't help.

So basically the question is: is there something special I should do that is not mentioned in documentation to run tests in Xcode? Or is there another way to get "detailed test reports" for iOS?

**Comment by bartekpacia** [comment]
Hi @aqrc, this is unfortunate

Actually, the video I recorded in the article required a small hack... so sorry about it! The hack is described [in this issue](https://github.com/leancodepl/patrol/issues/1342) - it should be pretty easy to do it yourself:

> What worked was removing the #ifdefs for PATROL_ENABLED - once all were removed, the tests run as seen on the video above.

You have to make these changes to your local copy of Patrol's source files – in my case it's `~/.pub-cache/hosted/pub.dev/patrol-2.2.0/ios/Classes`.

And you remove `#if`s and respective `#endif`s from these files:

![Screenshot 2023-08-09 at 11 29 51 AM](https://github.com/leancodepl/patrol/assets/40357511/f25a9218-7062-446b-9316-8c29812bc803)

Again, so sorry about it.

**Comment by aqrc** [comment]
Thank you for the quick reply! No worries, I probably should've ask earlier, just was curious if I could achieve it myself 😄
Also I somehow missed this issue even though I searched for `PATROL_ENABLED` 😬

Regarding the hack, I'll probably won't do it without a great need and will wait for a proper solution, but it's nice to know that there is such an option!

Thank you again!
